src/Command/Util/ApigeeEdgeManagementCliService.php

```diff
@@ -0,0 +1,330 @@
+        $this->httpClient->post($url, [
+          'body' => $body,
+          'auth' => [$email, $password],
+          'headers' => [
+            'Accept' => 'application/json',
+            'Content-Type' => 'application/json',
+          ],
+        ]);
+      }
+    }
+    catch (TransferException $exception) {
+      $this->handleHttpClientExceptions($exception, $io, $t, $url, $org, $email);
+      return;
+    }
+  }
+
+  /**
+   * Check to see if role exists.
+   *
+   * @param string $org
+   *   The Edge org to create the permissions in.
+   * @param string $email
+   *   The email of an Edge user with org admin role to make Edge API calls.
+   * @param string $password
+   *   The password of an Edge user with org admin role to make Edge API calls.
+   * @param string $base_url
+   *   The base url of the Edge API.
+   * @param string $role_name
+   *   The role name to add the permissions to.
+   *
+   * @return bool
+   *   Returns true if the role exists, or false if it doesn't.
+   *
+   * @throws \GuzzleHttp\Exception\TransferException
+   */
+  public function doesRoleExist(string $org, string $email, string $password, string $base_url, string $role_name) {
+    $url = $base_url . '/o/' . $org . '/userroles/' . $role_name;
+    try {
+      $this->httpClient->get($url, [
+        'auth' => [$email, $password],
+        'headers' => ['Accept' => 'application/json'],
+      ]);
+    }
+    catch (ClientException $exception) {
+      if ($exception->getCode() == 404) {
+        // Role does not exist.
+        return FALSE;
+      }
+      // Any other response was an exception.
+      throw $exception;
+    }
+    // If 200 response, role exists.
+    return TRUE;
+  }
+
+  /**
+   * Validate the Apigee Edge org connection settings.
+   *
+   * @param \Symfony\Component\Console\Style\StyleInterface $io
+   *   The IO interface of the CLI tool calling the method.
+   * @param callable $t
+   *   The translation function akin to t().
+   * @param string $org
+   *   The Edge org to connect to.
+   * @param string $email
+   *   The email of an Edge user with org admin role to make Edge API calls.
+   * @param string $password
+   *   The password of an Edge user email to make Edge API calls.
+   * @param string $base_url
+   *   The base url of the Edge API.
+   *
+   * @return bool
+   *   Return true if the Edge API can be called, false if it cannot.
+   */
+  public function isValidEdgeCredentials(StyleInterface $io, callable $t, string $org, string $email, string $password, string $base_url) {
+    $url = $base_url . '/o/' . $org;
+    try {
+      $response = $this->httpClient->get($url, [
+        'auth' => [$email, $password],
+        'headers' => ['Accept' => 'application/json'],
+      ]);
+    }
+    catch (TransferException $exception) {
+      $this->handleHttpClientExceptions($exception, $io, $t, $url, $org, $email);
+      return FALSE;
+    }
+
+    $body = json_decode($response->getBody());
```

Comment:
`JSON_THROW_ON_ERROR` would be nice here, but requires php 7.3 :(

Reply:
Agree, but we would need to up the PHP version allowed for our module, I don't think we should restrict it that much yet